src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -149,16 +149,31 @@ internal class TracerProviderSdk : TracerProvider
                 }
                 else
                 {
-                    var activitySources = new Dictionary<string, bool>(StringComparer.OrdinalIgnoreCase);
+                    var activitySources = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
 
-                    foreach (var name in sources)
+                    foreach (var source in sources)
                     {
-                        activitySources[name] = true;
+                        activitySources[source.Name] = source.Version;
                     }
 
                     // Function which takes ActivitySource and returns true/false to indicate if it should be subscribed to
                     // or not.
-                    listener.ShouldListenTo = (activitySource) => activitySources.ContainsKey(activitySource.Name);
+                    listener.ShouldListenTo = (activitySource) =>
+                    {
+                        if (activitySources.ContainsKey(activitySource.Name))
+                        {
+                            // retrieving from dictionary to check version
+                            var version = activitySources[activitySource.Name];
+                            if (string.IsNullOrWhiteSpace(version))
+                            {
+                                return true;
+                            }
+
+                            return version.Equals(activitySource.Version, StringComparison.OrdinalIgnoreCase);
```

Comment:
I'm not sure what would happen here.

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.version
Can we consider parsing the string to an actual `Version` and then use its comparison operators? So we can support things like "AddSource(library1, Version minVersion, Version maxVersion)", with version being optional (i.e subscrivbe to all versions)

Reply:
thanks for the review. Just updated